Sprint Challenge PR

josharmantrout91 opened the pull request:
Pulling together principles from Intermediate React methods to convert the Lambda Times site from vanilla JS to React.

josharmantrout91 committed:
Answers.md

- added Answers to Answers.md

josharmantrout91 committed:
Card Render

-passed props from Content.js -> Cards.js -> Card.js and have successful data render of passed props.

josharmantrout91 committed:
Tab Data Render and Filter Functionality

 - props passed from Content.js -> Tabs.js -> Tab.js and rendering correctly
 - cards filter correctly based on selected Tab

josharmantrout91 committed:
Styled Components

- completed refactor of TopBar.js using Styled Components
- completed refactor of Header.js using Styled Components
- added PropTypes to Cards.js, Card.js, Tabs.js, and Tab.js
- all MVP tasks complete @clintKunz

clintKunz commented:
## MVP
- [x] Go through the Tabs, Tab, Cards, and Card components following the instructions, and passing data and props to get the tabs and cards to appear on the screen.
- [x] Once the Tabs and Cards are rendering to the screen complete the changeSelected and filterCards functions in the Content component.
- [x] Make sure all of your props being passed are validated using PropTypes.
- [x] Find the TopBar and Header components. Convert these two components to Styled Components. You should not have any className props when you are finished.

## Stretch
- [ ] Re-factor the app, so that it uses ALL styled components. There should be no className props on any component. To truly test this, delete the CSS file.
- [ ] You will find a Carousel component in your Content folder. Complete this component, rendering a functional carousel. Add this component between your Tabs and Cards components within the Content component. Added challenge: make it so that there is infinite scroll to the right and the left.
- [ ] Add a login and an HOC. Make it so that when a user clicks on the login button at the top, a login modal is shown (Use React-strap). Have a user login, validating the login credentials on the localStorage. Add a Higher Order Component that wraps the Content component, only allowing it to render once a user has logged in.

## What Looks Great
- Great work overall
- Code is well formatted and easy to read

## What Could Be Improved
- See 3 warnings in console. 2 for unique key props and 1 for a failed propType.
- Need more commits. You should have one commit for every one change.

## Questions for Student
- Explain the details of a Higher Order Component

## Rating (1-3)
**2**

josharmantrout91 committed:
Error and Warning Debug

- fixed PropType error in Cards.js.
- removed unused variable { Component } from App.js and Cards.js
- added keys to Card and Tab in render method to remove key warning

clintKunz reviewed (approved):
Very nice!
